crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -1668,10 +1666,27 @@ fn generate_tangents_for_mesh(mesh: &Mesh) -> Result<Vec<[f32; 4]>, GenerateTang
     Ok(mikktspace_mesh.tangents)
 }
 
+/// Correctly scales and renormalizes an already normalized `normal` by the scale determined by its reciprocal `scale_recip`
+fn scale_normal(normal: Vec3, scale_recip: Vec3) -> Vec3 {
+    // This is basically just `normal * scale_recip` but with the added rule that `0. * anything == 0.`
+    // This is necessary because one of the components of `scale_recip` is infinite
```

Comment:
```suggestion
    // This is basically just `normal * scale_recip` but with the added rule that `0 * anything == 0`
    // This is necessary because components of `scale_recip` may be infinities, which do not multiply to zero
```